tests/integration/test_gitignore_patterns.py

```
@@ -0,0 +1,286 @@
+class TestGitignoreFileStructure:
+    """Tests for .gitignore file structure and syntax."""
+    
+    @pytest.fixture
+    def gitignore_path(self) -> Path:
+        """Return path to .gitignore file."""
+        return Path(__file__).parent.parent.parent / ".gitignore"
+    
+    @pytest.fixture
+    def gitignore_content(self, gitignore_path: Path) -> str:
+        """Load .gitignore file content."""
+        with open(gitignore_path, 'r', encoding='utf-8') as f:
```

Ruff (0.14.6)

Unnecessary mode argument

Remove mode argument

(UP015)